src/main/java/org/openjdk/jextract/impl/OutputFactory.java

```diff
@@ -258,6 +259,14 @@ public Void visitFunction(Declaration.Function funcTree, Declaration parent) {
             }
         }
 
+        // return type could be a function pointer type
+        Type.Function returnFunc = Utils.getAsFunctionPointer(funcTree.type().returnType());
```

Comment:
Did you check the corner case of a function pointer having a function pointer as its return type?

Reply:
While we recurse for pointer to function parameter types, we did recurse for return type. That's being fixed here.

For pointer to functions, we didn't recurse even for arguments (and also for return types).

For example, for

void (*func)(void (*f)(int));


we generate FI for "func" but not for "f".

The current fix case is seen in standard library (signal in signal.h). We can deal with recursing args and return types for function pointer types in a separate fix.